File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/purchace-invoice-line-dto.ts
import { EntityDto } from "../entity-dtos/entity-dto";
import { ProductDto } from "./product-dto";
import { PurchaceInvoiceDto } from "./purchace-invoice-dto";

export interface PurchaceInvoiceLineDto extends EntityDto<number>{
    productId: number;
    barCode?: string;
    invoiceNumber: number;
    quantity: number;
    unitPrice: number;
    totalPrice: number;
    invoice?: PurchaceInvoiceDto;
    product?: ProductDto;
}

File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/PurchaceInvoiceLineDto.cs
using ProjectPos.Services.EntityDtos;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectPos.Services.DTOs
{
    public class PurchaceInvoiceLineDto : EntityDto<int>
    {
        public int ProductId { get; set; }
        public int InvoiceNumber { get; set; }
        public int Quantity { get; set; }

        [Column(TypeName = "double(12, 2)")]
        public double UnitPrice { get; set; }

        [Column(TypeName = "double(12, 2)")]
        public decimal TotalPrice { get; set; }

        [ForeignKey("InvoiceNumber")]
        public PurchaceInvoiceDto? Invoice { get; set; }

        [ForeignKey("ProductId")]
        public ProductDto? Product { get; set; }
    }
}
